src/Component/AbstractComponent.php

```diff
@@ -266,7 +266,9 @@ protected function evaluateChildNodes(RenderingContextInterface $renderingContex
             if ($childNode instanceof EmbeddedComponentInterface) {
                 continue;
             }
-            $evaluatedNodes[] = $childNode->evaluate($renderingContext);
+            $childNodeCopy = clone $childNode;
```

Comment:
Might the problem be solved by `$evaluatedNodes[] = $childNode->evaluate(clone $renderingContext);` instead? It shouldn't be necessary to clone the node and the arguments every time - it should be enough to clone the context when evaluating (since the composition of the node and arguments doesn't change; only the context changes).